src/adsp_config.c

```diff
@@ -169,8 +169,8 @@ static uint32_t parse_uint32_hex_key(const toml_table_t *table, struct parse_ctx
  * @param error code, 0 when success
  * @return default, parsed, or UINT32_MAX value
  */
-static int parse_uint32_key(const toml_table_t *table, struct parse_ctx *ctx, const char *key,
-			    int64_t def, int *error)
+static uint32_t parse_uint32_key(const toml_table_t *table, struct parse_ctx *ctx, const char *key,
+				 int64_t def, int *error)
```

Comment:
Can we also update the doxygen to reflect the usage. i.e. on error a 0 is returned and ```error``` is set. Fwiw strtol has a similar calling convention except it sets ```errno```. TBH in parsing functions its always better to return the parsing status and have clients pass in an pointer for the parsing result (the unit32_t in this case)

Reply:
>Can we also update the doxygen

Done

>strtol has a similar calling convention except it sets errno

In `strtol` API, there is hard to remember about error code on the contrary to this api, when it's hard to forget about it 😄 

> TBH in parsing functions its always better to return the parsing status and have clients pass in an pointer for the parsing result (the unit32_t in this case)

It's a common convention, but  I don't understand the reason why it is so.
Impossibility to create the memory corruption of following struct fields is a strong argument for this API, at least from my point of view.
Additional profit is less lines of source code, when error code should be bubbled up there is a field size mismatch.
See:

```
	foo->bazz_u8 = (uint8_t) parse_uint32_key(src, &status);
	if (status)
		return status;
```

vs

```
        uint32_t temp;
	status= parse_uint32_key(src, &temp); /* memory corruption when  foo->bazz_u8 is accessed directly */
                                              /* notice, that data type may not be a part of a field name */
	if (status)
		return status;
	foo->bazz_u8 = (uint8_t)temp;
```